crates/librqbit_core/src/magnet.rs

```diff
@@ -45,20 +65,52 @@ impl Magnet {
 
 impl std::fmt::Display for Magnet {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(
-            f,
-            "magnet:?xt=urn:btih:{}&tr={}",
-            self.info_hash.as_string(),
-            self.trackers.join("&tr=")
-        )
+        if let (Some(id20), Some(id32)) = (self.id20, self.id32) {
+            write!(
+                f,
+                "magnet:?xt=urn:btih:{}?xt=urn:btmh:1220{}&tr={}",
+                id20.as_string(),
+                id32.as_string(),
+                self.trackers.join("&tr=")
+            )
+        } else if let Some(id20) = self.id20 {
+            write!(
+                f,
+                "magnet:?xt=urn:btih:{}&tr={}",
+                id20.as_string(),
+                self.trackers.join("&tr=")
+            )
+        } else if let Some(id32) = self.id32 {
+            write!(
+                f,
+                "magnet:?xt=urn:btmh:1220{}&tr={}",
+                id32.as_string(),
+                self.trackers.join("&tr=")
+            )
+        } else {
```

Comment:
please don't panic in Display trait. If you restructure the code like I suggested above, let it just print a magnet without hashes